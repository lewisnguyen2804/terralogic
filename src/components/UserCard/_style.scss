// PROFILE PICTURE & NAME
.name-image {
    margin: 40px 0 40px 0;
    display: flex;
    flex-direction: row;
    align-items: center;
    .user-image-change {
        position: relative;
        .user-image {
            width: 60px;
            height: 60px;
            border-radius: 50%;
            margin-right: 24px;
        }
        .user-image-change-icon {
            position: absolute;
            top: 0;
            margin-left: 35px;
            margin-top: -17px;
            left: 0;
            cursor: pointer;
            &:hover {
                opacity: 0.7;
            }
        }
    }
    .user-name {
        margin-top: 10px;
        @extend .bold-font;
    }
}