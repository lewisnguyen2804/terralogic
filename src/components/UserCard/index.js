import React, { Component } from 'react'
import ChangeUserImage from '../../assets/images/edit_photo.svg';

export default class UserCard extends Component {
    render() {
        const {
            name = 'Unknown',
            handleClick = {},
            image = 'http://api.terralogic.ngrok.io/-MDo7lIKOI7gn0-DNdNr_1596502631300_users-vector-icon-png_260862.jpg'
        } = this.props;
        return (
            <div className="name-image">
                <div className="user-image-change">
                    <img className="user-image"
                        alt="user-img"
                        src={image} />
                    <img className="user-image-change-icon"
                        src={ChangeUserImage}
                        alt="change-user-img"
                        onClick={handleClick} />
                </div>
                <h2 className="user-name">{name}</h2>
            </div>
        )
    }
}
