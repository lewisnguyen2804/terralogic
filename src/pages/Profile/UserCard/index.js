import React, { Component } from 'react'
import ChangeUserImage from '../../../assets/images/edit_photo.svg';

export default class UserCard extends Component {
    render() {
        return (
            <div className="name-image">
                <div className="user-image-change">
                    <img className="user-image"
                        alt="user-img"
                        src={this.props.image} />
                    <img className="user-image-change-icon"
                        src={ChangeUserImage}
                        alt="change-user-img"
                        onClick={this.props.handleClick} />
                </div>
                <h2 className="user-name">{this.props.name}</h2>
            </div>
        )
    }
}
