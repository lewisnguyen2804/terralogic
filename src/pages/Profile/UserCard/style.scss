@import '../../../style.scss';

// PROFILE PICTURE & NAME
.name-avatar {
    margin: 40px 0 40px 0;
    display: flex;
    flex-direction: row;
    align-items: center;
    .user-avatar {
        width: 60px;
        height: 60px;
        border-radius: 50%;
        margin-right: 24px;
    }
    .user-name {
        margin-top: 10px;
        @extend .bold-font;
    }
}