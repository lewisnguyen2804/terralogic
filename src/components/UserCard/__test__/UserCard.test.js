import React from 'react'
import { shallow } from 'enzyme'
import UserCard from '../';

describe('Test UserCard', () => {
    it('Test what?', () => {
        const userCard = shallow(<UserCard  />);
        expect(userCard).toMatchSnapshot()
    });
});