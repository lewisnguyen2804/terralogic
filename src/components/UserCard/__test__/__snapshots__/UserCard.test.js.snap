// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test UserCard Test what? 1`] = `ShallowWrapper {}`;
